playwright/src/test/java/com/microsoft/playwright/TestRequestContinue.java

```diff
@@ -75,18 +75,14 @@ void shouldOverrideRequestUrl() throws ExecutionException, InterruptedException
   @Test
   @Disabled("resume() method is now asynchronous")
   void shouldNotAllowChangingProtocolWhenOverridingUrl() {
-    PlaywrightException[] error = {null};
     page.route("**/*", route -> {
-      try {
+      PlaywrightException e = assertThrows(PlaywrightException.class, () -> {
         route.resume(new Route.ResumeOptions().setUrl("file:///tmp/foo"));
-      } catch (PlaywrightException e) {
-        error[0] = e;
-        route.resume();
-      }
+      });
+      assertTrue(e.getMessage().contains("New URL must have same protocol as overridden URL"), e.getMessage());
+      route.resume();
```

Comment:
I think this is correct due to the route handler being executed only once and not multiple times - previously was expecting at least one exceptions for all interceptions, now for each intercept we except an exception

Reply:
Sounds good.